craft_platforms/_distro.py

```diff
@@ -89,14 +89,27 @@ def _ensure_bases_comparable(
 
     def __eq__(self, other: object, /) -> bool | NotImplementedType:
         if isinstance(other, DistroBase):
-            return (
-                self.distribution == other.distribution and self.series == other.series
+            other_distro = other.distribution
+            other_series = other.series
+        elif isinstance(other, BaseName):
+            other_distro = other.name
+            other_series = other.version
+        elif isinstance(other, tuple):
+            other_distro = other[0]
+            other_series = other[1]
+        else:
+            return NotImplemented
```

Comment:
I think this is starting to duplicate `_get_version()`, but extends it to raises a `NotImplemented` error.

It also is starting to duplicate the if/else blocks in `_ensure_base_compatible()` to get the distro name.

This could be a good opportunity for refactoring to drop some duplicated code.

Reply:
Good point! Done!